src/plugins/data_source/server/plugin.ts

```diff
@@ -129,7 +129,8 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
       router,
       dataSourceService,
       cryptographyServiceSetup,
-      authRegistryPromise
+      authRegistryPromise,
+      customApiSchemaRegistryPromise
```

Comment:
do we know how this got missed at the first place?

Reply:
We need to add more tests for MDS to capture things like this, and I have added the only test for the test connection router. Can we fix this issue first then add more tests?